### PR TITLE
User authentication

### PR DESCRIPTION
**Pull request by Subhan636**

### CODE REVIEW

File: models/user.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const UserSchema = new Schema({
+    username: { type: String, required: true, unique: true },
+    password: { type: String, required: true }
+    // Add other user fields as needed
+});
+
+module.exports = mongoose.model('User', UserSchema);
```
Comment (gtaEPIC) [inline models/user.js:11]: Duplicate file. Other files is userRegistration.js

**Review by gtaEPIC — CHANGES_REQUESTED** [review]
https://github.com/gtaEPIC/COMP229-Deskify/pull/27/files#r1404752827

authController.js should reference the data from [userRegistraion.js](https://github.com/gtaEPIC/COMP229-Deskify/blob/main/models/userResgistration.js) since it's already built

File: controllers/authController.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+const User = require('../models/user');
```
Comment (gtaEPIC) [inline controllers/authController.js:1]:
```suggestion
const User = require('../models/userResgistration');
```
If you delete `user.js` and then change this, it should be better?
I'll still have to fully test this as well

**Comment by gtaEPIC** [comment]
The routes were also not added into [/config/app.js](https://github.com/gtaEPIC/COMP229-Deskify/blob/main/config/app.js)

**Review by imirza7 — APPROVED** [review]

File: controllers/authController.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+
+exports.user_login = async (req, res) => {
+    try {
+        const user = await User.findOne({ username: req.body.username });
+        if (user && bcrypt.compareSync(req.body.password, user.password)) {
```
Comment (gtaEPIC) [inline controllers/authController.js:8]: This doesn't check the password

Comment (gtaEPIC) [reply controllers/authController.js:8]: Because user registration didn't already have proper password hashing, that would be why

**Review by gtaEPIC — APPROVED** [review]

**Review by imirza7 — COMMENTED** [review]
All good!

**Pull request merged**